src/item.cpp

```diff
@@ -4132,6 +4132,10 @@ bool item::gunmod_compatible( const item& mod, bool alert ) const
     } else if( get_free_mod_locations( mod.type->gunmod->location ) <= 0 ) {
         msg = string_format( _( "Your %1$s doesn't have enough room for another %2$s mod." ), tname().c_str(), _( mod.type->gunmod->location.c_str() ) );
 
+    } else if( ammo_current() == "generic_no_ammo" ) {
+        msg = string_format( _( "You drain the charge from your %s before trying to modify it." ), tname().c_str() );
+        return true;
```

Comment:
The condition never gets tripped at all, it still gets caught at the next one down which checks for ammo_remaining > 0.
I got hung up on getting the condition to trigger, I didn't get to the point of actually integrating it properly. I don't want to short circuit all the other checks, just bypass the ammo remaining one.